scripts/git/prune_old_tags.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Deletes all but the input number of tags and corresponding releases, starting
+# with the oldest tags.
+
+KEEP_TAGS_COUNT="$1"
+UPSTREAM_REMOTE="${UPSTREAM_REMOTE:-upstream}"
+DRY_RUN="${DRY_RUN:-0}"
+
+if [[ -z "$KEEP_TAGS_COUNT" ]]; then
+  echo "Must specify the number of tags to keep"
+  exit 1
+fi
+
+if [[ "$KEEP_TAGS_COUNT" -le 0 ]]; then
+  echo "Cannot delete all remaining tags"
+  exit 1
+fi
+
+git fetch --all --tags
+TOTAL_TAGS_COUNT=$(git tag | wc -l)
+
+if [[ "$KEEP_TAGS_COUNT" -ge "$TOTAL_TAGS_COUNT" ]]; then
+  echo "Only $TOTAL_TAGS_COUNT exist, so nothing to delete"
+  exit 1
+fi
+
+DELETE_TAGS_COUNT=$((TOTAL_TAGS_COUNT-KEEP_TAGS_COUNT))
```

Comment:
Another suggestion for script ergonomics: how about `echo`ing the `TOTAL_TAGS_COUNT` and `DELETE_TAGS_COUNT` variables?

That would also be a good spot to ask for confirmation (e.g. https://stackoverflow.com/questions/1885525/how-do-i-prompt-a-user-for-confirmation-in-bash-script)